gap/SnAnUnknownDegree.gi

```diff
@@ -0,0 +1,295 @@
+function(G, c, eps, N, groupIsOne, groupIsEq)
+    local result, R, S, prebolsteringElms, i, r, cr, cr2;
+    result := [];
+    R := Int(Ceil(7 / 4 * Log2(Float(eps ^ -1))));
+    S := 7 * N * R;
+    prebolsteringElms := [];
+    i := 0;
+    # find pre-bolstering elements
+    while i <= S and Length(prebolsteringElms) <= R do
+        r := PseudoRandom(G);
+        # test whether r is pre-bolstering
+        cr := c ^ r;
+        cr2 := c ^ (r ^ 2);
+        if not groupIsOne(Comm(cr, c))
+                and not groupIsEq(cr2, c)
+                and not groupIsEq(cr2, c ^ 2)
+                and groupIsOne(Comm(cr2, c))
+        then
+            Add(prebolsteringElms, r);
+        fi;
+        i := i + 1;
+    od;
+    # construct bolstering elements
+    for r in prebolsteringElms do
+        if groupIsOne((c ^ (r * c * r)
+                      * c ^ (r * c ^ (r ^ 2) * c)) ^ 3)
+        then
+            Add(result, c ^ 2 * r);
+        else
+            Add(result, cr);
+        fi;
+    od;
+    return result;
+end);
+
+# g: a cycle matching c of a group G
+# c: a 3-cycle of a group G
+# r: arbitrary element of a group G
+# The supports of c and c^(g^2) have exactly one point, say alpha, in common.
+# Let phi be an isomorphism from G to a natural alternating or symmetric group.
+# This function decides whether alpha is a fixed point of phi(r).
+BindGlobal("IsFixedPoint",
+function(g, c, r, groupIsOne, groupIsEq)
+    local
+        # respectively c ^ (g ^ i)
+        cg, cg2, cg3, cg4,
+        # temporary holder of H1, H2
+        temp,
+        # sets of elements of G
+        H1, H2, XX,
+        # helper functions
+        constructH1, constructH2, isElmPassingTest;
+    # Helper functions
+    constructH1 := function(c, cg, cg3, cg4)
+        local H1;
+        H1 := ListWithIdenticalEntries(5,0);
+        H1[1] := c ^ 2;
+        H1[2] := c ^ cg;
+        H1[3] := H1[2] ^ cg3;
+        H1[4] := H1[3] ^ cg3;
+        H1[5] := H1[4] ^ cg4;
+        return H1;
+    end;
+    constructH2 := function(c, cg, cg3, cg4)
+        local H2;
+        H2 := ListWithIdenticalEntries(5,0);
+        H2[1] := c;
+        H2[2] := cg;
+        H2[3] := H2[2] ^ cg3;
+        H2[4] := H2[3] ^ cg3;
+        H2[5] := H2[4] ^ cg4;
+        return H2;
+    end;
+    isElmPassingTest := function(x, H, groupIsOne)
+        local nrTrivialComm, h;
+        nrTrivialComm := 0;
+        for h in H do
+            if groupIsOne(Comm(x, h)) then
+                nrTrivialComm := nrTrivialComm + 1;
+            fi;
+            if nrTrivialComm >= 2 then
+                return false;
+            fi;
+        od;
+        return true;
+    end;
+    cg := c ^ g;
+    cg2 := cg ^ g;
+    cg3 := cg2 ^ g;
+    cg4 := cg3 ^ g;
+    H1 := constructH1(c, cg, cg3, cg4);
+    # Test whether an elm of the set X, here called XX, commutes with at least
+    # two elements of H1.
+    XX := ListWithIdenticalEntries(3, 0);
+    XX[1] := c ^ r;
```

Comment:
Instead of allocating a list here, just use three local vars? E.g x1, x2, x3.

Reply:
Done